Cargo.toml

```diff
@@ -20,9 +20,10 @@ edition = "2018"
 # when the default feature set is updated, verify that the `--features` flags in
 # `.github/workflows/ci.yaml` are updated accordingly
 default = ["curl-client", "middleware-logger", "encoding"]
-h1-client = ["http-client/h1_client"]
-curl-client = ["http-client/curl_client", "once_cell"]
-wasm-client = ["http-client/wasm_client"]
+h1-client = ["http-client/h1_client", "default-client"]
+curl-client = ["http-client/curl_client", "once_cell", "default-client"]
+wasm-client = ["http-client/wasm_client", "default-client"]
+default-client = []
```

Comment:
Since this is not an actual feature I suggest doing some `#[cfg_attr()]` in `lib.rs` to set `#[cfg(default-client)]` instead, maybe?

Reply:
I tried to do this, but nothing I find in the docs indicates it works and I can't find an invocation of cfg_attr that actually conditionally sets a new configuration flag like that. I checked out the hyper branch in order to try to learn from it, but I'm also not convinced it's doing what we want in that branch — if you add a random line like `const S: &str = "";` after the cfg_attr line but before the next cfg attribute macro, it doesn't compile.

I've seen other crates using undocumented cargo.toml features like this, so my guess is that this is still the way to do conditional features

Reply:
Huh, upon examining https://doc.rust-lang.org/reference/conditional-compilation.html it seems I am misusing it in that PR. I have no idea how that PR manages to compile...